src/core/server/metrics/integration_tests/server_collector.test.ts

```diff
@@ -17,10 +17,9 @@ import { executionContextServiceMock } from '../../execution_context/execution_c
 import { ServerMetricsCollector } from '../collectors/server';
 import { setTimeout as setTimeoutPromise } from 'timers/promises';
 
-const requestWaitDelay = 25;
+const requestWaitDelay = 35;
```

Comment:
Was this run against the flaky runner? 

In the failing test, we are explicitly waiting until all the requests have at least been reaching the handler, via 

https://github.com/elastic/kibana/blob/a7556f480f4921791f208fd6c5b63e5d0a939d3a/src/core/server/metrics/integration_tests/server_collector.test.ts#L97-L102

So I'm not quite sure just increase the delay where we were already waiting should really resolves the flakiness, because in theory, we should not get to the assertions because all requests have reached the server? So I'm suspecting the flakiness is appearing from elsewhere.

Maybe we should also be adding an `await delay(requestWaitDelay);` between `await hitSubject...` and `let metrics = await collector.collect();` to some times for the HAPI internals?

Reply:
> Was this run against the flaky runner?

AFAIK, the flaky test runner doesn't run jest tests.

> So I'm suspecting the flakiness is appearing from elsewhere

I think we need a playbook on how to investigate flakiness in CI when we can't reproduce the issue locally. FWIW, I tested the timing by decreasing the `requestWaitDelay` locally to ~5.

Reply:
So, after some digging it seems like the flakiness is because the requests aren't always being triggered with `supertest(hapiServer.listener).get('/disconnect).end()` before the assertions. 

The `/disconnect' router hangs on `new Promise((resolve_ => undefined)` which is kinda intentional I guess to setup a disconnection. 

Even though the [callback to `.end`] is optional, I've added one [anyway](https://github.com/elastic/kibana/pull/58623#discussion_r385136588) since we're not awaiting the request. That and the added `delay` should hopefully stabilize this rather tricky test.